From 614e9acaa5ef61d92a7d8a03d9262d126a67e98b Mon Sep 17 00:00:00 2001
From: Davide Depau <davide@depau.eu>
Date: Wed, 12 May 2021 07:01:43 +0200
Subject: [PATCH] esp32: Add touch_set_cycles

---
 ports/esp32/modesp32.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/ports/esp32/modesp32.c b/ports/esp32/modesp32.c
index 3ed534338..578addc00 100644
--- a/ports/esp32/modesp32.c
+++ b/ports/esp32/modesp32.c
@@ -64,6 +64,22 @@ STATIC mp_obj_t esp32_wake_on_touch(const mp_obj_t wake) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(esp32_wake_on_touch_obj, esp32_wake_on_touch);
 
+STATIC mp_obj_t esp32_touch_set_cycles(const mp_obj_t measure_in, const mp_obj_t sleep_in) {
+    mp_int_t measure = mp_obj_get_int(measure_in);
+    mp_int_t sleep = mp_obj_get_int(sleep_in);
+
+#if CONFIG_IDF_TARGET_ESP32
+    //Touch pad SleepCycle Time
+    SET_PERI_REG_BITS(SENS_SAR_TOUCH_CTRL2_REG, SENS_TOUCH_SLEEP_CYCLES, sleep, SENS_TOUCH_SLEEP_CYCLES_S);
+    //Touch Pad Measure Time
+    SET_PERI_REG_BITS(SENS_SAR_TOUCH_CTRL1_REG, SENS_TOUCH_MEAS_DELAY, measure, SENS_TOUCH_MEAS_DELAY_S);
+#else
+    touch_pad_set_meas_time(sleep, measure);
+#endif
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(esp32_touch_set_cycles_obj, esp32_touch_set_cycles);
+
 STATIC mp_obj_t esp32_wake_on_ext0(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
 
     if (machine_rtc_config.wake_on_touch) {
@@ -187,6 +203,7 @@ STATIC const mp_rom_map_elem_t esp32_module_globals_table[] = {
     { MP_ROM_QSTR(MP_QSTR_wake_on_ext0), MP_ROM_PTR(&esp32_wake_on_ext0_obj) },
     { MP_ROM_QSTR(MP_QSTR_wake_on_ext1), MP_ROM_PTR(&esp32_wake_on_ext1_obj) },
     #if CONFIG_IDF_TARGET_ESP32
+    { MP_ROM_QSTR(MP_QSTR_touch_set_cycles), MP_ROM_PTR(&esp32_touch_set_cycles_obj) },
     { MP_ROM_QSTR(MP_QSTR_raw_temperature), MP_ROM_PTR(&esp32_raw_temperature_obj) },
     { MP_ROM_QSTR(MP_QSTR_hall_sensor), MP_ROM_PTR(&esp32_hall_sensor_obj) },
     #endif
-- 
2.31.1

